rcl/test/rcl/test_graph.cpp

```diff
@@ -310,7 +313,218 @@ check_graph_state(
   }
 }
 
-/* Test graph queries with a hand crafted graph.
+typedef std::function<rcl_ret_t (const rcl_node_t *,
+  const char * node_name,
+  rcl_names_and_types_t *)> GetTopicsFunc;
+
+void expect_topics_types(const rcl_node_t * node, GetTopicsFunc &func, size_t num_topics, const char * topic_name) {
+  rcl_ret_t ret;
+  rcl_names_and_types_t nat{};
+  nat = rcl_get_zero_initialized_names_and_types();
+  ret = func(node, topic_name, &nat);
+  EXPECT_EQ(num_topics, nat.names.size);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  rcl_reset_error();
+  ret = rcl_names_and_types_fini(&nat);
+  rcl_reset_error();
+}
+
+/**
+ * This test creates publishers, subscribers, and services, then calls node graph functions
+ * from the perspective of each node.
+ * It verifies each node perceives the same graph.
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION), test_node_info_functions) {
+  std::string topic_name("/test_node_info_functions__");
+  std::chrono::nanoseconds now = std::chrono::system_clock::now().time_since_epoch();
+  topic_name += std::to_string(now.count());
+  RCUTILS_LOG_INFO_NAMED(ROS_PACKAGE_NAME, "Using topic name: %s", topic_name.c_str());
+  rcl_ret_t ret;
+  const rcl_guard_condition_t *graph_guard_condition =
+    rcl_node_get_graph_guard_condition(this->node_ptr);
+
+  const rcl_guard_condition_t *remote_graph_guard_condition =
+    rcl_node_get_graph_guard_condition(this->node_ptr);
```

Comment:
Should we pass `remote_node_ptr` here instead?

Reply:
Definitely